proposals/12147-mlmd-removal/protos/artifacts.proto

```diff
@@ -0,0 +1,258 @@
+  // nextPageToken field.
+  int32 page_size = 3;
+
+  // Sorting order in form of "field_name", "field_name asc" or "field_name desc".
+  // Ascending by default.
+  string sort_by = 4;
+
+  // A url-encoded, JSON-serialized filter protocol buffer (see
+  // [filter.proto](https://github.com/kubeflow/artifacts/blob/master/backend/api/filter.proto)).
+  string filter = 5;
+}
+
+message ListArtifactResponse {
+  // The list of artifacts returned.
+  repeated Artifact artifacts = 1;
+
+  // The total number of artifacts available. This field is not always populated.
+  int32 total_size = 2;
+
+  // A token to retrieve the next page of results, or empty if there are no
+  // more results in the list.
+  string next_page_token = 3;
+}
+
+// The fields here work the same as previous backend api calls
+message ListArtifactEventsRequest {
+  // Optional, filter event by a set of task_ids
+  // We can also likely just rely on filter for this and omit this field
+  repeated string task_ids = 1;
+  // Optional, filter event by a set of artifact_ids
+  // We can also likely just rely on filter for this and omit this field
+  repeated string artifact_ids = 2;
+  
+  // Optional. Only list events that have artifacts of this event type.
+  ArtifactEventType task_type = 3;
+  
+  string page_token = 4;
+  int32 page_size = 5;
+  string sort_by = 6;
+  string filter = 7;
+}
+
+message ListArtifactEventsResponse {
+  repeated ArtifactEvents events = 1;
+  int32 total_size = 2;
+  string next_page_token = 3;
+}
+
+message LogMetricRequest {
+  // Required. The metric to log.
+  Metric metric = 1;
+}
+
+message GetMetricRequest {
+  // Required. Task UUID that owns this metric
+  string task_id = 1;
+  // Required. Name of the metric
+  string name = 2;
+}
+
+message ListMetricsRequest {
+  // A page token to request the results page.
+  string page_token = 1;
+  // The number of metrics to be listed per page.
+  int32 page_size = 2;
+  // Sorting order in form of "field_name", "field_name asc" or "field_name desc".
+  string sort_by = 3;
+  // A url-encoded, JSON-serialized filter protocol buffer.
+  string filter = 4;
+}
+
+message ListMetricsResponse {
+  // The list of metrics returned.
+  repeated Metric metrics = 1;
+  // The total number of metrics available.
+  int32 total_size = 2;
+  // Token to retrieve the next page of results.
+  string next_page_token = 3;
+}
+
+enum ArtifactEventType {
+  INPUT = 0;
+  OUTPUT = 1;
+}
+message ArtifactEvents {
+  string id = 1;
+  string artifact_id = 2;
+  string task_id = 3;
+
+  ArtifactEventType type = 4;
+  google.protobuf.Timestamp created_at = 5;
+}
+
+enum MetricType {
+  METRIC_INPUT = 0;
+  METRIC_OUTPUT = 1;
+}
+
+message Metric {
+  // Required. Task UUID that owns this metric
+  string task_id = 1;
+  // Required. Name of the metric
+  string name = 2;
+  // Optional. Numeric value of the metric
+  double number_value = 3;
+  // Optional. JSON value of the metric
+  string json_value = 4;
```

Comment:
Are we able to use `google.protobuf.Struct` here like we do for pipeline_spec? It's a nicer experience where the user doesn't have to serialize the JSON object.

Reply:
I see the ClassificationMetric has a list. I just dislike have to serialize JSON twice for an HTTP request and we don't have validation on the field at the Proto level.

Reply:
Maybe it can be its own message type for ClassificationMetric and then use `oneof`.